File: backend/dataset/levelOrder/levelorder_19.py
class Solution:
    def levelOrder(self, root: TreeNode) -> List[List[int]]:
        self._levels = []
        self.get_level(root, 0)
        return self._levels
            
    def get_level(self, root, depth):
        if(not root):
            return
        if(depth == len(self._levels)):
            self._levels.append([])
        self._levels[depth].append(root.val)
        self.get_level(root.left, depth + 1)
        self.get_level(root.right, depth + 1)

File: backend/CDataset/levelOrder/levelOrder_19.cpp
class Solution {
public:
    class TreeNode{
        public:
        TreeNode* left;
        TreeNode* right;
        int val;
    };

    void helper(TreeNode*root,vector<vector<int>>&result,int level){
        if(root){
            if(level==result.size()){result.push_back({root->val});}
            else{result[level].push_back(root->val);}
            helper(root->left,result,level+1);
            helper(root->right,result,level+1);
        }
    }
    vector<vector<int>>levelOrder(TreeNode* root) {
        vector<vector<int>>result;
        int level=0;
        helper(root,result,0);
        return result;
    }
};
